src/shm_main.c

```diff
@@ -768,6 +831,282 @@ sr_shmmain_createunlock(int shm_lock)
     }
 }
 
+/**
+ * @brief Determine whether the connection specified by cid is alive.
+ *
+ * @param[in]  cid Connection ID to check
+ * @param[out] conn_alive Set to non-zero if the connection is alive, 0 otherwise.
+ * @return err_info, NULL on success.
+ */
+sr_error_info_t *
+sr_shmmain_check_conn_lock(sr_cid_t cid, int *conn_alive) {
+    sr_error_info_t *err_info = NULL;
+    struct flock fl;
+    int fd = -1;
+    char *path = NULL;
+    int rc = 0;
+    conn_list_entry *ptr = NULL;
+
+    assert(conn_alive);
+
+    *conn_alive = 0;
+
+    /* connection ID NONE is reserved and never used */
+    if (SR_CONN_ID_NONE == cid) {
+        SR_ERRINFO_INT(&err_info);
+        sr_errinfo_free(&err_info);
+        return NULL;
```

Comment:
I suppose the error should be returned, why hide it. But I would say, again, that an assert for this condition seems like the best solution, `0` should never appear there and the input is generated internally.

Reply:
Agreed.  I converted it to an assert.